controllers/componentdetectionquery_controller.go

```diff
@@ -86,6 +86,11 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 		return ctrl.Result{}, err
 	}
 
+	// If on KCP, requeue if the "kcp.dev/cluster" annotation has not been added yet
+	if req.ClusterName != "" && componentDetectionQuery.GetAnnotations()["kcp.dev/cluster"] == "" {
```

Comment:
Wondering if there is a predicate that ignores changes to annotations? What happens if kcp decides to update the annotation label 🤔

Reply:
Unlikely to happen, as the label is specific to the KCP workspace the resource is on. Plus, if a reconcile _was_ triggered, then our deletion logic would trigger and the old CDQ would get deleted.